scripts/dts/gen_defines.py

```diff
@@ -193,61 +302,382 @@ def relativize(path):
 
 
 def write_regs(node):
-    # Writes address/size output for the registers in the node's 'reg' property
+    # Writes address/size output for the registers in the node's 'reg'
+    # property. This is where the BASE_ADDRESS and SIZE macros come from.
 
-    def write_reg(reg, base_ident, val):
-        # Drop '_0' from the identifier if there's a single register, for
-        # backwards compatibility
-        if len(reg.node.regs) > 1:
-            ident = f"{base_ident}_{reg.node.regs.index(reg)}"
-        else:
-            ident = base_ident
+    if not node.regs:
+        return
 
-        out_node(node, ident, val,
-                 # Name alias from 'reg-names = ...'
-                 f"{str2ident(reg.name)}_{base_ident}" if reg.name else None)
+    # This maps a reg_i (see below) to the "primary" BASE_ADDRESS and SIZE
+    # macros used to identify it, which look like:
+    #
+    # DT_<PRIMARY_NODE_IDENTIFIER>_BASE_ADDRESS_<reg_i>
+    # DT_<PRIMARY_NODE_IDENTIFIER>_SIZE_<reg_i>
```

Comment:
Could maybe indent code examples by two spaces or the like to make them stand out better from the main text.